NuRadioReco/modules/io/rno_g/readRNOGDataMattak.py

```diff
@@ -0,0 +1,318 @@
+         
+         # filter runs/datasets based on 
+         if select_runs and not self.__select_run(dataset):
+            self.__skipped_runs += 1
+            continue
+         
+         self.__n_runs += 1
+         self._datasets.append(dataset)
+         self.__n_events_per_dataset.append(dataset.N())
+
+      # keeps track which event index is in which dataset
+      self._event_idxs_datasets = np.cumsum(self.__n_events_per_dataset)
+      self._n_events_total = np.sum(self.__n_events_per_dataset)
+      
+      self._time_begin = time.time() - t0
+      
+      
+   def __select_run(self, dataset):
+      """ Filter/select runs/datasets. Return True to select an dataset, return False to skip it """
+      if not imported_runtable:
+         return True
+      
+      # get first eventInfo
+      dataset.setEntries(0)
+      event_info = dataset.eventInfo()
+      
+      run_id = event_info.run
+      station_id = event_info.station
+      
+      run_info = self.__run_table.query(f"station == {station_id:d} & run == {run_id:d}")
+      
+      if not run_info["run_type"].values[0] in self.__run_types:
+         return False
+      
+      if self.__max_trigger_rate and run_info["trigger_rate"].values[0] * units.Hz > self.__max_trigger_rate:
+         return False
+      
+      return True
+
+
+   def __get_n_events_of_prev_datasets(self, dataset_idx):
+      """ Get number of events from previous dataset to correctly set pointer """
+      dataset_idx_prev = dataset_idx - 1
+      return int(self._event_idxs_datasets[dataset_idx_prev]) if dataset_idx_prev >= 0 else 0
+
+
+   @register_run()
+   def run(self):
+
+      for event_idx in range(self._n_events_total):
+         self.logger.debug(f"Processing event number {event_idx} out of total {self._n_events_total}")
+         t0 = time.time()
+
+         # find correct dataset
+         dataset_idx = np.digitize(event_idx, self._event_idxs_datasets)
+         dataset = self._datasets[dataset_idx]
+
+         event_idx_in_dataset = event_idx - self.__get_n_events_of_prev_datasets(dataset_idx)
+         dataset.setEntries(event_idx_in_dataset)  # increment iterator -> point to new event
+         
+         event_info = dataset.eventInfo()
+
+         skip = False
+         if self._selectors is not None:
+            for selector in self._selectors:
+               if not selector(event_info):
+                  skip = True
+         
+         if skip:
+            self.__skipped += 1
+            continue
+                     
+         evt = NuRadioReco.framework.event.Event(event_info.run, event_info.eventNumber)
+         station = NuRadioReco.framework.station.Station(event_info.station)
+
+         trigger = NuRadioReco.framework.trigger.Trigger(event_info.triggerType)
+         trigger.set_triggered()
+         trigger.set_trigger_time(event_info.triggerTime)
+         station.set_station_time(datetime.datetime.fromtimestamp(event_info.triggerTime))
```

Comment:
Little nitpick: We decided to use `astropy.time` in NuRadioMC instead of `datetime` .The reason is that astropy supports nanosecond precision. Doesn't matter too much here, but let's keep it consistent.

Reply:
Good point, I will change that. I am currently wondering: What should I do with `event.__event_time`?

Reply:
actually the setter takes care of this right?
```
def set_station_time(self, time):
        if time is None:
            self._station_time = None
            return
        if isinstance(time, datetime.datetime):
            time_strings = str(time).split(' ')
            self._station_time = astropy.time.Time('{}T{}'.format(time_strings[0], time_strings[1]), format='isot')
        else:
            if time.format == 'datetime':
                time_strings = str(time).split(' ')
                self._station_time = astropy.time.Time('{}T{}'.format(time_strings[0], time_strings[1]), format='isot')
            else:
                self._station_time = time
```

Reply:
why
```
        if time is None:
            self._station_time = None
            return
```

Reply:
Uh but when converting to `datetime.datetime` we loss the nanosecond ... This is bad ...

Reply:
> why

okay now I get it xD